client/client.go

```diff
@@ -0,0 +1,64 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"istio.io/manager/model"
+)
+
+type ManagerClient struct {
+	base             *url.URL
+	versionedAPIPath string
+	client           *http.Client
+}
+
+func NewManagerClient(base *url.URL, apiVersion string, client *http.Client) *ManagerClient {
+	trimmedBase := base
+	if strings.HasSuffix(trimmedBase.Path, "/") {
+		trimmedBase.Path = trimmedBase.Path[:len(trimmedBase.Path)-1]
+	}
+	trimmedVersion := apiVersion
+	if strings.HasSuffix(trimmedVersion, "/") {
+		trimmedVersion = trimmedVersion[:len(trimmedVersion)-1]
+	}
+	if strings.HasPrefix(trimmedVersion, "/") {
+		trimmedVersion = trimmedVersion[1:]
+	}
+	if client == nil {
```

Comment:
Is this really necessary? Seems like it would complicate the API usage and it isn't much trouble the require the user to always pass in a `http.Client`. Also, without this extra check this function and the TrimSuffix/TrimPrefix changes this function can be written as:

```golang
return &ManagerClient{
    base:             strings.TrimSuffix(base.Path, "/"),
    versionedAPIPath: strings.TrimPrefix(strings.TrimSuffix(apiVersion, "/"), "/"),
}
```